packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -0,0 +1,211 @@
+///       duration: Duration(seconds: 1),
+///       builder: (BuildContext context, double size, Widget child) {
+///         return IconButton(
+///           iconSize: size,
+///           color: Colors.blue,
+///           icon: child,
+///           onPressed: () {
+///             setState(() {
+///               targetValue = targetValue == 24.0 ? 48.0 : 24.0;
+///             });
+///           },
+///         );
+///       },
+///       child: Icon(Icons.aspect_ratio),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [AnimatedBuilder], which builds custom animations that are controlled
+///    by a manually managed [AnimationController].
+///  * [ImplicitlyAnimatedWidget], which is a base class for [Widget]s that
+///    automatically animate any changes to their property values.
+class TweenAnimationBuilder<T> extends ImplicitlyAnimatedWidget {
+  /// Creates a [TweenAnimationBuilder].
+  ///
+  /// The properties [tween], [duration], and [builder] are required. The values
+  /// for [tween], [curve], and [builder] must not be null.
+  const TweenAnimationBuilder({
+    Key key,
+    @required this.tween,
+    @required Duration duration,
+    Curve curve = Curves.linear,
+    @required this.builder,
+    this.animationStatusListener,
+    this.child,
+  }) : assert(tween != null),
+       assert(curve != null),
+       assert(builder != null),
+       super(key: key, duration: duration, curve: curve);
+
+  /// Defines the target value for the animation.
+  ///
+  /// When the widget first builds, the animation runs from [Tween.begin] to
+  /// [Tween.end], if [Tween.begin] is non-null. A new animation can be
+  /// triggered at anytime by providing a new [Tween] with a new [Tween.end]
+  /// value. The new animation runs from the current animation value (which may
+  /// be [Tween.end] of the old [tween], if that animation completed) to
+  /// [Tween.end] of the new [tween]. The [Tween.begin] value is ignored except
+  /// for the initial animation that is triggered when the widget builds for the
+  /// first time.
+  ///
+  /// Any (subclass of) [Tween] is accepted as an argument. For example, to
+  /// animate the height or width of a [Widget], use a [Tween<double>], or
+  /// check out the [ColorTween] to animate the color property of a [Widget].
+  ///
+  /// Any [Tween] provided must have a non-null [Tween.end] value.
+  ///
+  /// ## Ownership
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [Tween].
+  /// Once a [Tween] instance has been passed to [TweenAnimationBuilder] its
+  /// properties should not be accessed or changed anymore as they may have been
+  /// modified by the [TweenAnimationBuilder]. If you need to change the
+  /// [Tween], create a **new instance** with the new values.
+  ///
+  /// It is good practice to never store a [Tween] provided to a
+  /// [TweenAnimationBuilder] in an instance variable.
+  final Tween<T> tween;
```

Comment:
Could this be an `Animatable<T>`? It doesn't look like `Tween.chain()` can be used without relaxing the type.

Reply:
It looks like `chain()` is most commonly used with a CurveTween (to add a curve before composing a tween that interpolates between another type), but since this widget already has support for curves I don't think this is actually a problem. (this widget doesn't need to cover every use-case anyway)